tls/s2n_cipher_suites.c

```diff
@@ -1219,6 +1220,18 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                 }
             }
 
+            /* For TLS1.3 when PSKs are present, the server must consider the hash algorithm associated with the chosen PSK
+             * when choosing a cipher suite. Server MUST reject any cipher suite without a matching hash algorithm and 
+             * continue to the next candidate. 
+             * */     
+            if (conn->actual_protocol_version >= S2N_TLS13 && conn->psk_params.chosen_psk != NULL) {
+                s2n_hmac_algorithm chosen_psk_hmac_alg;
```

Comment:
chosen_psk_hmac_alg = { 0 }

(there's a couple places you miss this in the tests too)